File: server/utils/wubrgify.py
def wubrgify(color_string):
    """Parse colors in WUBRG(C) order; removes duplicates and misordering"""
    res = ''
    for c in ['W', 'U', 'B', 'R', 'G']:
        res += c if c in color_string else ''
    return res if res else "C" # return "C" for colorless if W, U, B, R, G all not in color string

File: client/src/utils/wubrgify.ts
export function WUBRGify(colors: string) {
  let result: string = "";
  // WUBRGify the colors
  const _colors: string[] = ["W", "U", "B", "R", "G"];
  for (const color of _colors) {
    if (colors.includes(color)) result += color;
  }

  return result;
}
